src/viam/components/movement_sensor/movement_sensor.py

```diff
@@ -112,20 +112,21 @@ async def get_readings(self, *, extra: Optional[Dict[str, Any]] = None, timeout:
                 orientation: Orientation
             }
         """
-        ((pos, alt), lv, av, la, comp, orient) = await asyncio.gather(
-            self.get_position(extra=extra, timeout=timeout),
-            self.get_linear_velocity(extra=extra, timeout=timeout),
-            self.get_angular_velocity(extra=extra, timeout=timeout),
-            self.get_linear_acceleration(extra=extra, timeout=timeout),
-            self.get_compass_heading(extra=extra, timeout=timeout),
-            self.get_orientation(extra=extra, timeout=timeout),
-        )
-        return {
-            "position": pos,
-            "altitude": alt,
-            "linear_velocity": lv,
-            "angular_velocity": av,
-            "linear_acceleration": la,
-            "compass": comp,
-            "orientation": orient,
+        functions = {
+            "position": self.get_position,
+            "linear_velocity": self.get_linear_velocity,
+            "angular_velocity": self.get_angular_velocity,
+            "linear_acceleration": self.get_linear_acceleration,
+            "compass": self.get_compass_heading,
+            "orientation": self.get_orientation,
         }
+        readings = {}
+        for property, function in functions.items():
```

Comment:
My only issue with doing it this way is that we're now awaiting each function in series instead of running them concurrently.

I think this can be refactored back to the original code, but with `return_exceptions=True` (see [documentation](https://docs.python.org/3.11/library/asyncio-task.html#asyncio.gather))

This would result in something like:

```python3
        (pos, lv, av, la, comp, orient) = await asyncio.gather(
            self.get_position(extra=extra, timeout=timeout),
            self.get_linear_velocity(extra=extra, timeout=timeout),
            self.get_angular_velocity(extra=extra, timeout=timeout),
            self.get_linear_acceleration(extra=extra, timeout=timeout),
            self.get_compass_heading(extra=extra, timeout=timeout),
            self.get_orientation(extra=extra, timeout=timeout),
            return_exceptions=True,
        )
        return {
            "position": pos[0] if not isinstance(pos, Exception) else GeoPoint(),
            "altitude": pos[1] if not isinstance(pos, Exception) else 0,
            "linear_velocity": lv if not isinstance(lv, Exception) else Vector3(),
            "angular_velocity": av if not isinstance(av, Exception) else Vector3(),
            "linear_acceleration": la if not isinstance(la, Exception) else Vector3(),
            "compass": comp if not isinstance(comp, Exception) else 0,
            "orientation": orient if not isinstance(orient, Exception) else Orientation(),
        }
```

This way, we address @maximpertsov issue of not dropping keys (and also be in line with RDK, which does not drop keys), and we continue to keep things parallelized.